src/engine.rs

```diff
@@ -16,14 +16,20 @@ use crate::{
     AtomEdit, FmtDiff,
 };
 
+pub(crate) enum ExtraInfo<'a> {
+    Explanation(&'a mut Vec<(AtomEdit, Option<RuleName>)>),
+    Edits(&'a mut Vec<AtomEdit>, &'a mut Vec<AtomEdit>),
```

Comment:
I guess it might be helpful to document in a comment what these two vecs are used for (i.e. the difference between them).

Reply:
Does e57a49bc6e57ce7de124de06ed20350c48d81141 clarify things sufficiently? I'm not really sure what else to say about them...

Reply:
I'd prefer some doc-comments with the reasoning for splitting this up. Perhaps @zimbatm can chime in to provide further context here.

Reply:
Can you add the doc-comments please? Having a small text that says /// Edits takes a list of spacing and indentation changes. Or something like that.

Reply:
Fair enough, is the comment in 7a8e061ec4635a0a2c5a32d5176012a15e6d481d ok?

Reply:
I thiink it'd be reasonable to add a small comment for `Explanation` as well.

Reply:
Ok, added in 75b22ba87dc9e5f82d98fc1767e3589239e7b4c6.